aws-api-appsync/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequest.java

```diff
@@ -288,19 +297,23 @@ public Builder variable(@NonNull String key, String type, Object value) {
             Objects.requireNonNull(this.operation);
             Objects.requireNonNull(this.responseType);
 
-            if (modelClass != null) {
-                // Derive modelSchema and selectionSet from modelClass.
+            if (modelClass == null && modelSchema == null) {
+                throw new AmplifyException("Both modelSchema and modelClass cannot be null", "");
+            }
+
+            if (modelSchema == null) {
+                // Derive modelSchema from modelClass if not available
                 modelSchema = ModelSchema.fromModelClass(this.modelClass);
+            }
```

Comment:
This can be removed now right?  Since an exception is thrown above if `modelSchema` is null, I think this code block is unreachable now?

```suggestion
```

Reply:
It doesn't actually work this way -- `modelSchema` is still potentially null here. The control flow is a little screwy. I'll simplify it, so the behavior is more clear.